docs/book/src/developer/repository-layout.md

```diff
@@ -1,3 +1,175 @@
 # Repository Layout
 
-This page is still being written - stay tuned!
+This page covers the repository structure and details about the directories in CAPI.
+
+```
+projectapi
+└───api
+│   │   v1alpha3
+│   │   v1alpha4
+│   │   v1beta1
+│   
+└───bootstrap
+│   │   kubeadm
+│   │   util
+│  
+└───cmd
+│   │   clusterctl
+│
+└───config
+│   │   certmanager
+│   │   crd
+│   │   default
+│   │   manager
+│   │   rbac
+│   │   webhook
+│  
+└───controllers
+│   
+└───controlplane
+│   │   kubeadm
+│
+└───docs
+│   
+└───errors
+│   
+└───exp
+│   
+└───feature
+│   
+└───hack
+│   
+└───internal
+│   
+└───scripts
+│
+└───test
+│   │   e2e
+│   │   framework
+│   └───infrastructure
+│       │   container
+│       │   docker
+│   
+└───util
+│   
+└───version
+│   
+└───webhooks
+│   
+└───main.go
+│   
+└───Makefile
+│
+└───tilt-settings.json
+```
+
+### API
+
+[~/api](https://github.com/kubernetes-sigs/cluster-api/tree/main/api)
+
+This folder is used to store types and their related resources present in CAPI core. It includes things like API types, spec/status definitions, condition types, simple webhook implementation, autogenerated, deepcopy and conversion files. Some examples of Cluster API types defined in this package include Cluster, ClusterClass, Machine, MachineSet, MachineDeployment and MachineHealthCheck.
+
+API folder has subfolders like v1alpha3, v1alpha4 and v1beta1. These packages contain API schema definitions for the respective API groups.
```

Comment:
```suggestion
API folder has subfolders for each supported API version.
```
Let's drop references to specific API version otherwise this will require maintenance